src/coreclr/inc/gcinfotypes.h

```diff
@@ -678,8 +678,8 @@ void FASTCALL decodeCallPattern(int         pattern,
 #define NORMALIZE_SIZE_OF_STACK_AREA(x) ((x)>>2)
 #define DENORMALIZE_SIZE_OF_STACK_AREA(x) ((x)<<2)
 #define CODE_OFFSETS_NEED_NORMALIZATION 1
-#define NORMALIZE_CODE_OFFSET(x) (x)   // Instructions are 2/4 bytes long in Thumb/ARM states,
```

Comment:
Matching change will need to be made in https://github.com/dotnet/diagnostics/tree/main/src/shared/gcdump so that GC info dumping in SOS works.

There is a challenge though: SOS wants to support all runtimes, so the SOS copy will need to work for both old and new GC info encoding - we will need to switch based on the current runtime version somehow.

Reply:
Also, same problem exists for https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/Amd64

Reply:
> Also, same problem exists for https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/Amd64

There is no corresponding location for arm[64]. Do we use Amd64 for all cases?
It is the RISC arches that are more interesting, since the size changes.

Reply:
I think there are no changes needed on x64 for the uses of GcInfo that do enumeration. The offsets are different, but that is a matter of `FindSafePoint(offset)` kind of use, as the offset would need to no longer do `-1` adjustment in nonleaf cases. I do not see such use in ReadyToRun thing.  
Basically - for the use like `foreach(safePoint in ...) {foreach(slot in SlotsAt(safePoint)} ` , everything should work as before.

Reply:
The RISC is more interesting as GcInfo deserializer would need to denormalize code offsets based on GcInfo version. Not just the offsets used in safepoints, there are other places.

I would expect that GcInfo deserializer that wants to support different runtimes already needs to figure the version of GCInfo. This is not the first time it changes.

Reply:
Actually safepoint enumeration is also affected. The indices use `CeilOfLog2` bits and that is relative to the normalized method length. 
(shorter method will have fewer safepoints, so index can be shorter and that is scaled with code offset normalization.

Reply:
The diagnostics/SOS GcInfo deserializer looks like just a copy of the one in the runtime. It is version-aware (can handle v1 and v2) and gets the version from the supplied token.

We may just need to copy/paste the new version once this change goes in.  

Logged a follow-up issue for that: https://github.com/dotnet/diagnostics/issues/4795